File: Data Structures & Algorithms/reverse-a-linked-list/submission-2.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func reverseList(head *ListNode) *ListNode {
    // recursion version
    if head == nil {
        return nil
    }

    prev := head
    if head.Next != nil {
        prev = reverseList(head.Next)
        head.Next.Next = head
    }
    head.Next = nil 

    return prev
}

File: Data Structures & Algorithms/maximum-twin-sum-of-a-linked-list/submission-2.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        fast, slow = head, head
        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next
        
        prev = None
        while slow:
            tmp = slow.next
            slow.next = prev
            prev = slow
            slow = tmp

        ans = 0
        pointer = head
        while prev:
            curr = prev.val + pointer.val
            ans = max(ans, curr)
            prev = prev.next
            pointer = pointer.next
        
        return ans